### DISCUSSION TITLE
The current connection holds no SharePoint context. Please use one of the Connect-PnPOnline commands which uses the -Url argument to connect.

### DISCUSSION THREAD

**Discussion by faisaleem-na** [pnp/powershell#3500] [Q&A]
### Discussed in https://github.com/pnp/powershell/discussions/2465

<div type='discussions-op-text'>

<sup>Originally posted by **ItsTerryDuckett** October 14, 2022</sup>
I have an Azure Runbook with a system managed identity. I'm using the following code as a simple test:
```
$conn = Connect-PnPOnline -Url "https://tenant.sharepoint.com/sites/sitename" -ManagedIdentity -ReturnConnection
$list = Get-PnPList -Identity "listid" -Connection $conn

```
The "Connect-PnPOnline -ManagedIdentity" works fine and connects. However, the next line, "$list = Get-PnPList -Identity "listid" -Connection $conn" returns the following error:
_"The current connection holds no SharePoint context. Please use one of the Connect-PnPOnline commands which uses the -Url argument to connect."_

If I switch the runbook back to using a "runas" account, it works fine. 
If I try and use "Connect-PnPOnline -ManagedIdentity -Url $url" the above error occurs</div>

**Comment by AndersRask** [comment]
Try and add -ValidateConnection parameter (added in 1.12 or so) to Connect-PnPOnline to get an error. Without that it will not always throw any kind of error
